ext/ddtrace.stub.php

```diff
@@ -136,9 +144,59 @@ public function getStartTime(): int {}
          * @return SpanLink Get a pre-populated SpanLink object with the current span's trace and span IDs
          */
         public function getLink(): SpanLink {}
+
+        /**
+         * @return Returns the span id as zero-padded 16 character hexadecimal string.
+         */
+        public function hexId(): string {}
     }
 
     class RootSpanData extends SpanData {
+        /**
+         * @var string The origin site of the trace. Propagated through distributed tracing by default.
+         */
+        public string $origin;
+
+        /**
+         * @var array A hashset of keys which are propagated from meta, if present.
+         */
+        public array $propagatedTags = [];
+
+        /**
+         * @var int The currently active sampling priority.
+         */
+        public int $samplingPriority = \DD_TRACE_PRIORITY_SAMPLING_UNKNOWN;
+
+        /**
+         * @var int The unmodified sampling priority as inherited directly through distributed tracing.
+         */
+        public int $propagatedSamplingPriority;
+
+        /**
+         * @var string The original tracestate minus datadog specific tags, as it will be propagated to upstream
+         * distributed tracing targets.
+         */
+        public string $tracestate;
+
+        /**
+         * @var array A list of datadog specific tags, which will be propagated to upstream distributed tracing
+         * targets as part of the tracestate. Some known keys are not included here, but directly extracted, e.g. origin.
+         */
+        public array $tracestateTags = [];
+
+        /**
+         * @var string The unique identifier of the parent span.
+         * Assignment of an invalid id will unset the parent id.
+         * This variable cannot be accessed by reference.
+         */
+        public string $parentId;
+
+        /**
+         * @var string The unique identifier for the trace id, as a zero-padded 16 character hexadecimal string.
+         * Assignment of an invalid id will reset the trace id to the default trace id.
+         * This variable cannot be accessed by reference.
+         */
+        public string $traceId = "";
```

Comment:
Let's say I have a span and want to retrieve the current 128-bit trace ID (zero-padded 32-character hexdec string). How would it be done?

💭 Else, maybe it makes sense to give the possibility to retrieve through a method (e.g., `hexTradeId`)?
```c
PHP_METHOD(DDTrace_SpanData, hexTraceId) {
	ddtrace_span_data *span = OBJ_SPANDATA(Z_OBJ_P(ZEND_THIS));
	RETURN_STR(ddtrace_trace_id_as_hex_string(span->root ? span->root->trace_id : (ddtrace_trace_id){ .low = span->span_id, .high = 0 }));
}
```
 which would prevent the code duplication associated with either:
- The concatenation of `_dd.p.tid` (if any, else zeros) and `RootSpanData->traceId`?
- The conversion from the decimal trace ID returned by `DDTrace\trace_id()` to a zero-padded 32-character hexdec string?

Reply:
Eih, it actually is the 32-character id, it's a mistake in the description.